File: java/src/main/java/itp/itp1_2/ITP1_2_D.java
package itp.itp1_2;

import java.util.Scanner;

public class ITP1_2_D {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    
    int W = sc.nextInt();
    int H = sc.nextInt();
    int x = sc.nextInt();
    int y = sc.nextInt();
    int r = sc.nextInt();
    sc.close();

    if (0 <= x - r && x + r <= W && 0 <= y - r && y + r <= H) {
      System.out.println("Yes");
    } else {
      System.out.println("No");
    }
  }
}

File: kotlin/src/itp/itp1_2/ITP1_2_D.kt
package itp.itp1_2

import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val W = sc.nextInt()
    val H = sc.nextInt()
    val x = sc.nextInt()
    val y = sc.nextInt()
    val r = sc.nextInt()
    println(if (x+r in (r..W) && y+r in (r..H)) "Yes" else "No")
}